packages/debug/src/browser/console/debug-console-session.ts

```diff
@@ -15,40 +15,32 @@
  ********************************************************************************/
 
 import throttle = require('@theia/core/shared/lodash.throttle');
-import { injectable, inject, postConstruct } from '@theia/core/shared/inversify';
 import { DebugProtocol } from 'vscode-debugprotocol/lib/debugProtocol';
 import { ConsoleSession, ConsoleItem } from '@theia/console/lib/browser/console-session';
 import { AnsiConsoleItem } from '@theia/console/lib/browser/ansi-console-item';
 import { DebugSession } from '../debug-session';
-import { DebugSessionManager } from '../debug-session-manager';
 import URI from '@theia/core/lib/common/uri';
 import { ExpressionContainer, ExpressionItem } from './debug-console-items';
 import { Severity } from '@theia/core/lib/common/severity';
 
-@injectable()
 export class DebugConsoleSession extends ConsoleSession {
```

Comment:
Why have you made this class no longer `injectable`? To facilitate extensibility, we generally want to add `injectability` rather than removing t?

Reply:
Right, didn't really think about that. I made `DebugConsoleSession` injectable again and added a `DebugConsoleSessionProvider` that directly sets the internal `debugSession` to the provided value.